File: lat2_c++/memory.cpp
#include <iostream>
#include <string>

using namespace std;

class memory : public hardware{
private:
	string frequency;
	string memorySize;
	string suppCuda;

public:
	memory(){
		this->frequency = "";
		this->memorySize = "";
		this->suppCuda = "";	
	}

	memory(string frequency, string memorySize, string suppCuda){
		this->frequency = frequency;
		this->memorySize = memorySize;
		this->suppCuda = suppCuda;
	}

	void setfrequency(string frequency){
		this->frequency = frequency;
	}

	void setmemorySize(string memorySize){
		this->memorySize = memorySize;
	}

	void setsuppCuda(string suppCuda){
		this->suppCuda = suppCuda;
	}


	string getfrequency(){
		return frequency;
	}

	string getmemorySize(){
		return memorySize;
	}

	string getsuppCuda(){
		return suppCuda;
	}

	void printMemory(){
		cout << "Frequency : ";
		cout << getfrequency() << endl;
		cout << "Memory : ";
		cout << getmemorySize() << endl;
		cout << "Cuda : ";
		cout << getsuppCuda() << endl;
	}

	~memory(){}
};

File: lat2_java/memory.java
class memory extends hardware{
    private String frequency;
    private String memorySize;
    private String supportsCuda;
    
    public memory(){
    }

    public memory(String frequency, String memorySize, String supportsCuda) {
        this.frequency = frequency;
        this.memorySize = memorySize;
        this.supportsCuda = supportsCuda;
    }

    public String getFrequency(){
        return frequency;
    }

    public void setFrequency(String frequency){
        this.frequency = frequency;
    }

    public String getMemorySize(){
        return memorySize;
    }

    public void setMemorySize(String memorySize){
        this.memorySize = memorySize;
    }

    public String getSupportsCuda(){
        return supportsCuda;
    }

    public void setSupportsCuda(String supportsCuda){
        this.supportsCuda = supportsCuda;
    }

    public void printMemory(){
        System.out.print("Frequency : ");
        System.out.println(getFrequency());
        System.out.print("Memory : ");
        System.out.println(getMemorySize());
        System.out.print("Cuda : ");
        System.out.println(getSupportsCuda());

    }
}
